### PR TITLE
Rely on Asciidoctor official parser

### LINKED ISSUE #3: Consider using Asciidoc Java parser

### PR DESCRIPTION
**Pull request by fbiville**
Fixes #3

### CODE REVIEW

**Comment by fbiville** [comment]
I only found a hacky way to unescape XML entities.
I still need to find a way to preserve line breaks for attributes such as:
```asciidoc
:query: MATCH (p:Product)-[:PART_OF]->(:Category)-[:PARENT*0..]-> +
 (:Category {categoryName:$category}) +
 RETURN p.productName as product +
```
cc @Mogztter

File: src/main/java/com/neo4j/sandbox/updater/MetadataReader.java [hunk 1]
```diff
@@ -1,56 +1,45 @@
...
         try (BufferedReader bufferedReader = new BufferedReader(path)) {
-            return bufferedReader.lines().collect(Collectors.toList());
+            return bufferedReader.lines().collect(Collectors.joining("\n"));
         }
     }
 
-    private String processQueryLine(String line) {
-        String strippedLine = line.replace(":query: ", "").replaceAll("^\\s+", "");
-        return strippedLine.substring(0, strippedLine.length() - "+".length());
+    private static String processQueryLine(String line) {
```
Comment (fbiville) [inline src/main/java/com/neo4j/sandbox/updater/MetadataReader.java:42]: I'd love to get rid of this :)

**Comment by Mogztter** [comment]
@fbiville you can use a passthrough macro:

```adoc
= Title
:query: pass:[MATCH (p:Product)-[:PART_OF]->(:Category)-[:PARENT*0..]-> + \
 (:Category {categoryName:$category}) + \
 RETURN p.productName as product]
```

`doc.getAttribute('query')` will return:


```
MATCH (p:Product)-[:PART_OF]->(:Category)-[:PARENT*0..]-> +
(:Category {categoryName:$category}) +
RETURN p.productName as product
```

I don't know the context so it's hard to suggest a solution but I would recommend to declare the query in a block rather than in an attribute:

```
[query]
....
MATCH (p:Product)-[:PART_OF]->(:Category)-[:PARENT*0..]->
(:Category {categoryName:$category})
RETURN p.productName as product
....
```

It will be easier to maintain, more readable and you will be able to preserve formatting.
You can use a custom block or an existing block (like `[source,cypher]`).

**Comment by fbiville** [comment]
Thanks @Mogztter for the suggestions!

# TL;DR
We want to keep this sandbox metadata hidden, so I guess attributes are the only way to go.
I think I can insert the `pass` macro instead of forcing every sandbox README to set it.

# NTL;DAR (Not Too Long; Did Actually Read 😛 )
Let me give some context here: when you start a new sandbox instance on https://sandbox.neo4j.com/, you have access to customized code samples to show you how to get started in the programming language of your choice.

Screenshot 2021-02-26 at 11 49 18

This customization relies on "hidden" README attributes, from each dataset repository in https://github.com/neo4j-graph-examples/. Here is an example from the [Twitter trolls dataset](https://raw.githubusercontent.com/neo4j-graph-examples/twitter-trolls/) [README](https://raw.githubusercontent.com/neo4j-graph-examples/twitter-trolls/main/README.adoc):
```markdown
[...]
:query: MATCH (u:User {screen_name: 'queenofthewo'})-[:POSTED]->(t:Tweet)-[:HAS_TAG]->(ht:Hashtag {tag:$tag}) +
RETURN t.created_str as createdTime +

:param-name: tag
:param-value: thingsdonebymistake
:result-column: createdTime
:expected-result: 2017-02-27 15:19:24
[...]
```
Based on this metadata, we will adapt the code sample [templates](https://github.com/neo4j-graph-examples/template/tree/master/code) and generate the resulting [code samples](https://github.com/neo4j-graph-examples/twitter-trolls/tree/main/code).
